braidpool-primitives/src/braid/mod.rs

```diff
@@ -85,6 +110,120 @@ impl Braid {
 }
 
 impl Braid {
+    // All pub(crate) functions go here!
+    #[inline]
+    pub(crate) fn load_bead_from_hash(&self, bead_hash: BeadHash) -> Result<&Bead, BeadLoadError> {
+        // This functions returns a bead from memory! Future DB work goes in here!
+
+        // TODO: Add in a check for whether a bead_hash is valid!
+
+        match self.loaded_beads_in_memory.get(&bead_hash) {
+            Some(bead) => Ok(bead),
+            None => Err(BeadLoadError::BeadPruned),
+        }
+    }
+}
+
+impl Braid {
+    fn update_parents_of_bead(&self, child_bead: &Bead) {
```

Comment:
We never need to update the parents of a bead.

When a Bead is created, its parents should be set in its constructor. If a new bead comes in, the Node should destroy the old Bead we were mining on, and create a new one (it will have different parents and possibly different transactions). So we basically have to recreate a whole new Bead from scratch instead of modifying an existing Bead.